last_seen.py

```diff
@@ -11,23 +11,39 @@ def seen_evt(self, event):
         events = ['quit', 'part', 'kick', 'join', 'rename', 'topic', 'message']
         if event.event_type in events:
             nick = event.meta['nick']
-            self.seen[nick] = (event.event_type, gmtime())
+            self.seen[nick] = (event.event_type, datetime.datetime.utcnow())
 
     @cmd
     def last_seen(self, event):
-        """
-        Returns when was given user last seen.
+        """Return when was given user last seen.
 
         Example: !last_seen <user>
         """
         args = event.args
         if len(args) < 1:
-            return "Name argument missing."
+            return 'Name argument missing.'
         user = args[0]
 
         if user not in self.seen:
-            return "This user was never here."
+            return 'This user has never been here.'
 
         evt, time = self.seen[user]
-        time = strftime("%H:%M %d.%m.%Y", time)
-        return "{0} was last seen: {1} GMT({2})".format(user, time, evt)
+        diff_time = datetime.datetime.utcnow() - time
+
+        if diff_time.days == 0:
+            hours = diff_time.seconds // 3600
+            minutes = (diff_time.seconds - (hours * 3600)) // 60
+
+            min_format = 'minute' if minutes == 1 else 'minutes'
+            if hours > 0:
+                hour_format = 'hour' if hours == 1 else 'hours'
+                msg = '{0} {1} and {2} {3}'.format(hours, hour_format,
+                                                   minutes, min_format)
+            elif minutes > 0:
+                msg = '{0} {1}'.format(minutes, min_format)
+            else:
+                msg = 'less than 1 minute'
+            return '{0} was last seen {1} ago'.format(user, msg)
+
+        time = time.strftime('%H:%M %d.%m.%Y')
```

Comment:
@Adman in my personal opinion the month should be 0-padded. Other than that I'd say day-first apparoach makes sense, since the day is what you are looking for anyway in this case (last event took place more than 24 hours ago).